edb/server/protocol/ai_ext.py

```diff
@@ -1205,53 +1251,133 @@ async def _start_openai_like_chat(
                     continue
 
                 if sse.data == "[DONE]":
+                    # mistral doesn't send finish_reason for tool calls
+                    if finish_reason == "unknown":
+                        event = (
+                            b'event: content_block_stop\n'
+                            + b'data: {"type": "content_block_stop",'
+                            + b'"index": ' + str(tool_index).encode() + b'}\n\n'
+                        )
+                        protocol.write_raw(event)
                     event = (
                         b'event: message_stop\n'
                         + b'data: {"type": "message_stop"}\n\n'
                     )
                     protocol.write_raw(event)
-                    continue
+                    break
```

Comment:
Hmm, @mmastrac any idea why this could be the case?  FWIW breaking explicitly on the client side is probably OK, but this might indicate a bug in the new http client code.

Reply:
Fixed this in https://github.com/edgedb/edgedb/pull/7984